tests/Integration/Behaviour/Features/Scenario/Cart/Adding/CartRule/add_cartrule.feature

```diff
@@ -113,7 +113,7 @@ Feature: Add cart rule in cart
   Scenario: Add cart rule which provides gift product and free shipping
     Given I create an empty cart "dummy_cart_1" for customer "testCustomer"
     And I select "US" address as delivery and invoice address for customer "testCustomer" in cart "dummy_cart_1"
-    And I add 2 products "Mug The best is yet to come" to the cart "dummy_cart_1"
+    And I update quantity of product "Mug The best is yet to come" in the cart "dummy_cart_1" to 2
```

Comment:
Be careful you replaced every single former use of `I add 2 products` with `I update quantity of product`

So I know that it was incorrect before since it always used the update command to add new product, but since we have both commands available now we should use them appropriately So when it's an empty cart or when it doesn't contain the added product we should use `I add 2 products`  This will allow us to validate its behaviour as well